app/src/main/java/org/schabi/newpipe/local/feed/notifications/NotificationWorker.kt

```
@@ -0,0 +1,172 @@
+            .build()
+        setForegroundAsync(ForegroundInfo(FeedLoadService.NOTIFICATION_ID, notification))
+    }
+
+    companion object {
+
+        private val TAG = NotificationWorker::class.java.simpleName
+        private const val WORK_TAG = App.PACKAGE_NAME + "_streams_notifications"
+
+        private fun areNotificationsEnabled(context: Context) =
+            NotificationHelper.areNewStreamsNotificationsEnabled(context) &&
+                NotificationHelper.areNotificationsEnabledOnDevice(context)
+
+        /**
+         * Schedules a task for the [NotificationWorker]
+         * if the (device and in-app) notifications are enabled,
+         * otherwise [cancel]s all scheduled tasks.
+         */
+        @JvmStatic
+        fun initialize(context: Context) {
+            if (areNotificationsEnabled(context)) {
+                schedule(context)
+            } else {
+                cancel(context)
+            }
+        }
+
+        /**
+         * @param context the context to use
+         * @param options configuration options for the scheduler
+         * @param force Force the scheduler to use the new options
+         * by replacing the previously used worker.
+         */
+        fun schedule(context: Context, options: ScheduleOptions, force: Boolean = false) {
+            val constraints = Constraints.Builder()
+                .setRequiredNetworkType(
+                    if (options.isRequireNonMeteredNetwork) {
+                        NetworkType.UNMETERED
+                    } else {
+                        NetworkType.CONNECTED
+                    }
+                ).build()
+
+            val request = PeriodicWorkRequest.Builder(
+                NotificationWorker::class.java,
+                options.interval,
+                TimeUnit.MILLISECONDS
+            ).setConstraints(constraints)
+                .addTag(WORK_TAG)
+                .setBackoffCriteria(BackoffPolicy.LINEAR, 30, TimeUnit.MINUTES)
```

I am still not so convinced about having a backoff criteria... I'd prefer to just show an error notification (like it's already being done) and let the periodic worker just run again after a while, ignoring the error.

---

Now `createWork()` return `Result.failure()` on error, not a `Result.retry()`

---

I removed it, since as you said it will never be triggered. There is a default backoff anyway, so it doesn't complain if we don't provide one (i.e. it's not a required parameter).